src/OpenLoco/Paint/PaintMiscEntity.cpp

```diff
@@ -0,0 +1,323 @@
+#include "../Interop/Interop.hpp"
+#include "../Localisation/StringIds.h"
+#include "../Map/Tile.h"
+#include "Paint.h"
+#include <cassert>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::Paint
+{
+    static void paintExhaustEntity(PaintSession& session, Exhaust* exhaustObject)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        if (dpi->zoom_level > 1)
+        {
+            return;
+        }
+        SteamObject* steamObject = exhaustObject->object();
+
+        uint8_t* edi = (exhaustObject->object_id & 0x80) == 0 ? steamObject->var_16 : steamObject->var_1A;
+        uint32_t imageId = edi[2 * exhaustObject->var_26];
+        imageId = imageId + steamObject->baseImageId + steamObject->var_0A;
+
+        if ((steamObject->var_08 & 8) != 0)
+        {
+            session.addToPlotListAsParent(imageId, { 0, 0, exhaustObject->z }, { 1, 1, 0 });
+        }
+        else
+        {
+            session.addToPlotListAsParent(imageId, { 0, 0, exhaustObject->z }, { 1, 1, 0 }, { -12, -12, exhaustObject->z });
+        }
+    }
+
+    static void paintRedGreenCurrencyEntity(PaintSession& session, MoneyEffect* moneyEffect)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        int8_t* wiggleYOffsets = reinterpret_cast<int8_t*>(0x4FAAC8);
+        if (dpi->zoom_level > 1)
+        {
+            return;
+        }
+        const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_green : StringIds::format_currency_expense_red_negative;
+        uint32_t currencyAmount = abs(moneyEffect->amount);
+        int8_t* yOffsets = &wiggleYOffsets[moneyEffect->wiggle];
+
+        session.addToStringPlotList(currencyAmount, stringId, moneyEffect->y, moneyEffect->z, yOffsets, moneyEffect->offsetX);
+    }
+
+    static void paintWindowCurrencyEntity(PaintSession& session, MoneyEffect* moneyEffect)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        int8_t* wiggleYOffsets = reinterpret_cast<int8_t*>(0x4FAAC8);
+        if (dpi->zoom_level > 1)
+        {
+            return;
+        }
+        const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_in_company_colour : StringIds::format_currency_expense_in_company_colour_negative;
+        uint32_t currencyAmount = abs(moneyEffect->amount);
+        int8_t* yOffsets = &wiggleYOffsets[moneyEffect->wiggle];
+        uint16_t companyColour = CompanyManager::getCompanyColour(moneyEffect->var_2E);
+
+        session.addToStringPlotList(currencyAmount, stringId, moneyEffect->y, moneyEffect->z, yOffsets, moneyEffect->offsetX, companyColour);
+    }
+
+    static void paintVehicleCrashParticleEntity(PaintSession& session, VehicleCrashParticle* particle)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        if (dpi->zoom_level != 0)
+        {
+            return;
+        }
+
+        loco_global<int32_t[5], 0x4FAAB4> vehicle_particle_base_sprites;
```

Comment:
This is a list of image offsets: { 3302, 3314, 3326, 3338, 3350 }. Could you translate it into an std::map of five std::arrays with constants for the relevant image ids? I think there will be 12 elements in each array.

Reply:
Updated in the next commit.